javascript/role-details.js

```diff
@@ -0,0 +1,294 @@
+            assignEmployeeDiv.classList.add("assign-employee");
+            let assignEmployeeDetails = document.createElement("div");
+            assignEmployeeDetails.classList.add("assign-employee-details");
+            let assignEmployeeImg = document.createElement("img");
+            assignEmployeeImg.src = "assets/images/profie-icon.png";
+            let employeeName = document.createElement("label");
+            employeeName.classList.add("check-employee");
+            employeeName.innerText = r.Name;
+            employeeName.setAttribute("for", "chekCheckBox" + a);
+            assignEmployeeDetails.appendChild(assignEmployeeImg);
+            assignEmployeeDetails.appendChild(employeeName);
+            let check = document.createElement("INPUT");
+            check.setAttribute("type", "checkbox");
+            check.classList.add("employee-check-box");
+            check.setAttribute('id', "chekCheckBox" + a);
+            let id = r.empNo;
+            check.onchange = function () {
+                checkUnassignedEmployees(this, id);
+            };
+            a++;
+            assignEmployeeDiv.appendChild(assignEmployeeDetails);
+            assignEmployeeDiv.appendChild(check);
+            document.getElementById("assign-employee-section").appendChild(assignEmployeeDiv);
+        }
+    })
+}
+let visible = false;
+function showUnassignedEmployees() {
+    if (visible) {
+        closeUnassignedEmployees();
+    }
+    else {
+        openUnassignedEmployees();
+    }
+    visible = !visible;
+}
+
+function openUnassignedEmployees() {
+    document.getElementById("assign-employee-section").style.display = "flex";
+}
+
+function closeUnassignedEmployees() {
+    document.getElementById("assign-employee-section").style.display = "none";
+}
+
+//check box checked
+function checkUnassignedEmployees(className, employeeID) {
+    if (unassignedEmployee.includes(employeeID)) {
+        unassignedEmployee.splice(unassignedEmployee.indexOf(employeeID), 1);
+    }
+    else {
+        unassignedEmployee.push(employeeID);
+    }
+}
+
+function addRoleForEmployee() {
+    // event.preventDefault;.
+    let inputRole = document.getElementById("role-name-input");
+    employee.forEach(r => {
+        if (unassignedEmployee.includes(r.empNo)) {
+            r.role = inputRole.value;
+        }
+    })
+    localStorage.setItem('employee', JSON.stringify(employee));
+
+}
+document.addEventListener("click", (event) => {
+    let target = event.target;
+    if (target.className != "asssign-employees-container" && target.className != "searchEmployee" && target.className != "assign-employee" && target.className != "assign-employee-details" && target.className != "check-employee" && target.className != "employee-check-box") {
+        if (document.querySelector(".asssign-employees-container").style.display == "flex") {
+            showUnassignedEmployees();
+        }
+    }
+})
+
+function addEditListner() {
```

Comment:
I hope we are using the same function in other js file. If the functionality is same can we move it to common.js. Else rename it accordingly.